### PR TITLE
feat: Make marketing collection rail header customizable

### PR DESCRIPTION
**Pull request by olerichter00**
This PR resolves [CX-3473] 

### Description

Make marketing collection rail header customizable.

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3473]: https://artsyproduct.atlassian.net/browse/CX-3473?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/Home/Components/MarketingCollectionRail.tsx [hunk 1]
```diff
@@ -47,37 +53,74 @@ export const MarketingCollectionRail: React.FC<MarketingCollectionRailProps> = (
...
-const artworksFragment = graphql`
-  fragment MarketingCollectionRail_viewer on Viewer
+export const homeFragment = graphql`
+  fragment MarketingCollectionRail_home on HomePage
+  @argumentDefinitions(heroImageVersion: { type: "HomePageHeroUnitImageVersion" }) {
+    heroUnits(platform: MOBILE) {
+      slug
+      title
+      subtitle
+      backgroundImageURL(version: $heroImageVersion)
```
Comment (olerichter00) [inline src/app/Scenes/Home/Components/MarketingCollectionRail.tsx:113]: Taking the title, subtitle, and header image from the hero unit to be able to customize it without updating the app.

Comment (MounirDhahri) [reply src/app/Scenes/Home/Components/MarketingCollectionRail.tsx:113]: good idea!

**Review by MounirDhahri — APPROVED** [review]

**Comment by MounirDhahri** [comment]
LGTM! thanks

**Comment by olerichter00** [comment]
> Do we need to remove the old image or keep it as a default image if no image is available 👀

This is a good question 🤔 Maybe we should just remove it so we can see if there is something wrong with the image. What do you think?

**Comment by MounirDhahri** [comment]
let's do it

**Pull request merged**